backend/hitas/services/owner.py

```diff
@@ -0,0 +1,64 @@
+from django.db import models
+from django.db.models import Count, OuterRef
+from django.db.models.functions import Coalesce
+
+from hitas.models.housing_company import RegulationStatus
+from hitas.models.owner import Owner, OwnerT
+from hitas.models.ownership import Ownership
+from hitas.utils import SQSum
+
+
+def obfuscate_owners_without_regulated_apartments() -> list[OwnerT]:
```

Comment:
Instead of doing so much data processing in python, wouldn't it be easier to let the db handle it?
(Increases db queries by 1, but less time is spent running python code)

```py
owners_qs = Owner.objects.annotate(...).filter(...)

obfuscated_owners: list[OwnerT] = list(owners_qs.values("name", "identifier", "email"))

owners_qs.update(
    name=None,
    identifier=None,
    valid_identifier=False,
    email=None,
    bypass_conditions_of_sale=True,
)
return obfuscated_owners
```

Reply:
Yes, this is much cleaner, and if I add an if-check before the queryset.update, it's the same amount of queries since the `obfuscated_owners` is evaluated once, and then the update is depending on if there are obfuscated owners.

```python
if obfuscated_owners:
    owners.update(...)
```